test/service_manager_test.dart

```diff
@@ -108,6 +108,101 @@ void main() {
       await _verifyExtensionStateInServiceManager(extensionName, true, 0.5);
     });
 
+    test('callService', () async {
+      dynamic _e;
+
+      final registeredService =
+          serviceManager.methodsForService['reloadSources'] ?? const [];
+      expect(registeredService, isNotEmpty);
+
+      try {
+        await serviceManager.callService('reloadSources',
+            isolateId: serviceManager.isolateManager.selectedIsolate.id);
+      } catch (e) {
+        _e = e;
+      }
+
+      expect(_e, isNull);
+    });
+
+    test('callService throws exception', () async {
+      // Service with less than 1 registration.
+      Exception exception;
+      Exception expectedException =
+          Exception('Expected one registered service for fakeMethod'
+              ' but found 0');
+
+      try {
+        await serviceManager.callService('fakeMethod');
+      } catch (e) {
+        exception = e;
+      }
+
+      expect(exception, isNotNull);
+      expect(exception.toString(), equals(expectedException.toString()));
+
+      // Service with more than 1 registration.
+      serviceManager.methodsForService.putIfAbsent('fakeMethod',
+          () => ['registration1.fakeMethod', 'registration2.fakeMethod']);
+      exception = null;
+      expectedException =
+          Exception('Expected one registered service for fakeMethod'
+              ' but found 2');
+
+      try {
+        await serviceManager.callService('fakeMethod');
+      } catch (e) {
+        exception = e;
+      }
+
+      expect(exception, isNotNull);
+      expect(exception.toString(), equals(expectedException.toString()));
+    });
+
+    test('callMulticastService', () async {
+      dynamic _e;
+
+      final registeredService =
+          serviceManager.methodsForService['reloadSources'] ?? const [];
+      expect(registeredService, isNotEmpty);
+
+      try {
+        await serviceManager.callMulticastService('reloadSources',
```

Comment:
nit: you can verify that a valid Result object is returned as well.

Reply:
done.